src/Symfony/Component/HttpClient/Response/AsyncResponse.php

```diff
@@ -0,0 +1,355 @@
+                        }
+
+                        if (!$stream->valid()) {
+                            break;
+                        }
+                    } catch (\Throwable $e) {
+                        $r->info['error'] = $e->getMessage();
+                        $r->response->cancel();
+
+                        yield $r => $chunk = new ErrorChunk($r->offset, $e);
+                        $chunk->didThrow() ?: $chunk->getContent();
+                        unset($asyncMap[$response]);
+                        break;
+                    }
+
+                    $chunk = $stream->current();
+
+                    if (!$chunk instanceof ChunkInterface) {
+                        throw new \LogicException(sprintf('A chunk passthru must yield instances of "%s", "%s" yielded.', ChunkInterface::class, get_debug_type($chunk)));
+                    }
+
+                    if (null !== $chunk->getError()) {
+                        // no-op
+                    } elseif ($chunk->isFirst()) {
+                        $e = $r->openBuffer();
+
+                        yield $r => $chunk;
+
+                        if (null === $e) {
+                            continue;
+                        }
+
+                        $r->response->cancel();
+                        $chunk = new ErrorChunk($r->offset, $e);
+                    } elseif ('' !== $content = $chunk->getContent()) {
+                        if (null !== $r->shouldBuffer) {
+                            throw new \LogicException('A chunk passthru must yield an "isFirst()" chunk before any content chunk.');
+                        }
+
+                        if (null !== $r->content && \strlen($content) !== fwrite($r->content, $content)) {
+                            $chunk = new ErrorChunk($r->offset, new TransportException(sprintf('Failed writing %d bytes to the response buffer.', \strlen($content))));
+                            $r->info['error'] = $chunk->getError();
+                            $r->response->cancel();
+                        }
+                    }
+
+                    if (null === $chunk->getError()) {
+                        $r->offset += \strlen($content);
+
+                        yield $r => $chunk;
+
+                        if (!$chunk->isLast()) {
+                            continue;
+                        }
+
+                        $stream->next();
+
+                        if ($stream->valid()) {
+                            throw new \LogicException('A chunk passthru cannot yield after an "isLast()" chunk.');
+                        }
+
+                        $r->passthru = null;
+                    } else {
+                        if ($chunk instanceof ErrorChunk) {
+                            $chunk->didThrow(false);
+                        } else {
+                            try {
+                                $chunk = new ErrorChunk($chunk->getOffset(), !$chunk->isTimeout() ?: $chunk->getError());
```

Comment:
Why is this class forcefully, silently converting non-error chunks to ErrorChunks? I would assume there would be a reason why would chunk not identify itself as ErrorChunk, but still return error message - that shouldn't be a concern of this class.

Reply:
This code is to provide compatibility with implementations of ChunkInterface that don't use ErrorChunk to signal errors (an independent implementation should not use `ErrorChunk` since it's `@internal`)
It does change the logic but just provide interop.